main/src/library/Reducible.flix

```diff
@@ -84,52 +84,62 @@ pub lawless class Reducible[t: Type -> Type] {
     ///
     /// Returns the first element of `t`.
     ///
-    pub def head(t: t[a]): a = ???
+    pub def head(t: t[a]): a =
+        Reducible.reduceLeft((acc, _) -> acc, t)
 
     ///
     /// Returns the last element of `t`.
     ///
-    pub def last(t: t[a]): a = ???
+    pub def last(t: t[a]): a =
+        Reducible.reduceLeft((_, a) -> a, t)
 
     ///
     /// Returns a list with the first element of `t`.
     ///
-    pub def init(t: t[a]): List[a] = ???
+    pub def init(t: t[a]): List[a] =
+        Reducible.reduceLeftTo((acc, _) -> acc, a -> a :: Nil, t)
 
     ///
-    /// Returns a list with the elements of `t` except the first.
+    /// Returns the tail of `t`.
     ///
-    pub def tail(t: t[a]): List[a] = ???
+    pub def tail(t: t[a]): List[a] =
+        Reducible.reduceLeftTo((k, a) -> ks -> k(a :: ks),  _ -> identity, t)(Nil)
```

Comment:
This should probably also be overwritten in the instance for Nel

Its hard for me to judge if the implementation is correct. Whats the intuition?

Reply:
I fear I may be super allergic to `foldRight` (due to the fear that it may not be tailrecursive) which would make this much more readable.
The intuition here is that the first element is dropped due to `_ -> identity` ignoring the argument. The rest simply builds a continuation function which is then immediately invoked with `(Nil)`.

Reply:
On it! :smile: